### ISSUE TITLE
Source control diff viewer displays an undo arrow that has no action for staged changes

### ISSUE THREAD

**Issue by amunger** [microsoft/vscode#165264]
Type: <b>Bug</b>

Make a change in a local repository and stage that change so that it's ready to commit.
Open the source control activity and view the change.
:bug: the arrow in the gutter doesn't do anything

VS Code version: Code - Insiders 1.73.0-insider (8fa188b2b301d36553cbc9ce1b0a146ccb93351f, 2022-11-01T15:30:14.729Z)
OS version: Windows_NT x64 10.0.22621
Modes:
Sandboxed: Yes

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-1065G7 CPU @ 1.30GHz (8 x 1498)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|undefined|
|Memory (System)|31.59GB (14.95GB free)|
|Process Argv|--crash-reporter-id 1202dc88-403d-4ebc-8393-af3551a8bd5b|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (33)</summary>

Extension|Author (truncated)|Version
---|---|---
Bookmarks|ale|13.3.1
tsl-problem-matcher|amo|0.6.2
vscode-eslint|dba|2.2.6
python-environment-manager|don|1.0.4
EditorConfig|Edi|0.16.4
prettier-vscode|esb|9.9.0
codespaces|Git|1.13.1
copilot|Git|1.56.7152
vscode-pull-request-github|Git|0.53.2022110210
vscode-test-explorer|hbe|2.21.1
svg|joc|1.4.22
prettify-json|moh|0.0.3
vscode-language-pack-de|MS-|1.73.11020948
vscode-dotnet-runtime|ms-|1.6.0
isort|ms-|2022.5.13061128
python|ms-|2022.19.13061131
vscode-pylance|ms-|2022.11.10
data-wrangler|ms-|0.0.461
jupyter|ms-|2022.10.110
jupyter-keymap|ms-|1.0.0
jupyter-renderers|ms-|1.0.12
vscode-jupyter-cell-tags|ms-|0.1.6
vscode-jupyter-slideshow|ms-|0.1.5
remote-containers|ms-|0.262.2
remote-ssh|ms-|0.91.2022103115
remote-wsl|ms-|0.72.0
js-debug-nightly|ms-|2022.10.1817
remote-explorer|ms-|0.1.2022103109
remote-repositories|ms-|0.22.0
test-adapter-converter|ms-|0.1.6
vscode-github-issue-notebooks|ms-|0.0.127
vscode-js-profile-flame|ms-|1.0.4
vscode-yaml|red|1.10.1

</details><details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627:30244334
vslsvsres303:30308271
pythonvspyl392:30422396
pythontb:30258533
pythonptprofiler:30281269
vshan820:30294714
pythondataviewer:30285072
vscod805cf:30301675
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593cf:30376535
pythonvs932:30404738
cppdebug:30492333
vsclangdf:30492506
c4g48928:30535728
dsvsc012:30540252
pylantcb52:30590116
vsccc:30564266
pyindex848:30577858
nodejswelcome1:30587009
gswce1:30598410
34cci188:30599831
pyind779cf:30600160

```

</details>

**Comment by amunger** [comment]
![image](https://user-images.githubusercontent.com/2019016/199562633-be405ab3-66dc-43b1-9d93-87e93aa91238.png)

**Comment by lszomoru** [comment]
@alexdima, @hediet do you know who owns the "arrow" action in the editor gutter?

**Comment by hediet** [comment]
@mingpan added them. Is this because the target file is readonly?

**Comment by Mingpan** [comment]
@hediet Yes I think it's very likely due to readonly file. I tried it with the fix (#165423) and cannot reproduce this bug there. Once a file is staged, viewing the change doesn't show the arrows anymore.

**Comment by alexdima** [comment]
@hediet Can you reproduce this? I'm having trouble reproducing, the arrow doesn't show for me when clicking on a modified file from the "Staged" section:


<img width="1017" alt="image" src="https://user-images.githubusercontent.com/5047891/204284580-9bc70afd-6085-4607-9864-351dba08632f.png">

**Comment by Mingpan** [comment]
Probably because the fix is already merged, so the bug (revert arrows show up for readonly files) doesn't reproduce anymore?

**Comment by alexdima** [comment]
Oh true, I totally missed that we had a PR to fix this :D. Thank you!
